seed/building_sync/building_sync.py

```diff
@@ -237,534 +154,185 @@ def export(self, property_state, process_struct=ADDRESS_STRUCT):
-                    return False
-                elif isinstance(new_node, dict):
-                    return self._set_node(new_path, new_node, value)
-                else:
-                    # can't recurse futher into new_node because it is not a dict
-                    break
-
-    def _set_compound_node(self, list_path, data, key_path_name, key_path_value, value_path_name,
-                           value):
+        return xmlschema.XMLSchema(schema_path)
+
+    def restructure_mapped_result(self, result, messages):
+        """Transforms the dict from applying a mapping into a more standardized structure
+        for SEED to store into a model
+
+        :param result: dict, the mapped values
+        :param messages: dict, dictionary for recording warnings and errors
+        :return: dict, restructured dictionary
         """
-        If the XML is a list of options with a key field at the same level as the value, then use
-        this method. The example belows show how the XML will be structured. To set the
-        Gross floor area, then pass in the following
-
-            _set_compound_node("...FloorAreas", "FloorAreaType", "Gross", "FloorAreaValue", 1000)
-
-        .. code:
-
-            <auc:FloorAreas>
-                <auc:FloorArea>
-                    <auc:FloorAreaType>Gross</auc:FloorAreaType>
-                    <auc:FloorAreaValue>25000</auc:FloorAreaValue>
-                </auc:FloorArea>
-                <auc:FloorArea>
-                    <auc:FloorAreaType>Net</auc:FloorAreaType>
-                    <auc:FloorAreaValue>22500</auc:FloorAreaValue>
-                </auc:FloorArea>
-            </auc:FloorAreas>
-
-        :param list_path: String, path to where the list of items start in the dictionary
-        :param data: Dict, data to act on
-        :param key_path_name: String, name of the element to contrain check on
-        :param key_path_value: String, name of the value of the element to check on
-        :param value_path_name: String, name of the element that will be set
-        :param value: undefined, Value to set
-        :return: Boolean
+        measures = []
+        for measure in result['measures']:
+            if measure['category'] == '':
+                messages['warnings'].append(f'Skipping measure {measure["name"]} due to missing category')
+                continue
+
+            measures.append(measure)
+
+        scenarios = []
+        for scenario in result['scenarios']:
+            # process the scenario meters (aka resource uses)
+            meters = {}
+            for resource_use in scenario['resource_uses']:
+                meter = {}
+                meter['source'] = Meter.BUILDINGSYNC
+                meter['source_id'] = resource_use['source_id']
+                meter['type'] = resource_use['type']
+                meter['units'] = resource_use['units']
+                meter['is_virtual'] = scenario['is_virtual']
+                meter['readings'] = []
```

Comment:
You could also set these key values during the construction of the meter dict. I'd imagine it'd be faster to set them all at once if possible, but that's just my guess.
```
                meter = {
                    'source': Meter.BUILDINGSYNC,
                    ...
                }
```